aten/src/ATen/native/cuda/Normalization.cu

```diff
@@ -60,22 +60,28 @@ std::tuple<Tensor, Tensor> batch_norm_stats_cuda(const Tensor& self, double epsi
 
 Tensor batch_norm_elemt_cuda(const Tensor& self, const Tensor& weight, const Tensor& bias,
                              const Tensor& mean, const Tensor& invstd, double epsilon) {
+  auto output = at::empty_like(self);
+  return batch_norm_elemt_cuda_out(output, self, weight, bias, mean, invstd, epsilon);
+}
+
+Tensor batch_norm_elemt_cuda_out(Tensor& output, const Tensor& self, const Tensor& weight, const Tensor& bias,
```

Comment:
Return a Tensor& here and I think your problem will go away. You might need to adjust your inner code to directly return output.

Reply:
Is returning a reference to a local variable (tensor) safe?

Reply:
I'm not sure I'm understanding your suggestion. Do you mean replacing `Tensor` as the return type of `batch_norm_elemt_cuda_out` and `batch_norm_elemt_cuda` by `Tensor&`?

Reply:
Even if we change return type, in theory, the error should persist: `Oct 09 19:39:07 /var/lib/jenkins/workspace/aten/src/ATen/native/cuda/Normalization.cu(67): error: cannot overload functions distinguished by return type alone
`

Reply:
> Is returning a reference to a local variable (tensor) safe?

No it is not. However, the convention is that `_out` functions return the reference for out that was passed in. It should look something like:

```
Tensor& foo_out(Tensor& output, const Tensor& input) {
  // ...
  return output;
}
```

Reply:
> Even if we change return type, in theory, the error should persist

I'm not sure I follow. There might be other errors in your code, but your error message is consistent with this C++ code:

```
// Auto-generated in NativeFunctions.h
Tensor& foo_out(Tensor& output, const Tensor& input);

// Your implementation
Tensor foo_out(Tensor& output, const Tensor& input) {
  ...
}
```

This is exactly the situation a C++ compiler will complain about an overload distinguished only in return type.

By the way, you can find out exactly what signature ATen is expecting by inspecting the generated native functions header.

Reply:
@ezyang Thanks for explanation! I now understand you, I'll try the suggested return type for the `_out`-variant.